src/main/java/de/muenchen/isi/domain/service/AbfrageStatusService.java

```diff
@@ -302,4 +299,116 @@ public UUID getAbfrageId(final MessageHeaders messageHeaders) throws EntityNotFo
         }
         return abfrageId;
     }
+
+    /**
+     * Ermittelt mögliche Statusübergänge basierend auf Benutzer Authorities und aktuellen Status der Abfrage.
+     *
+     * @param id vom Typ {@link UUID} um die Abfrage zu finden
+     * @return Liste von {@link TransitionModel} welche möglich sind
+     * @throws EntityNotFoundException falls die Abfrage nicht gefunden wird
+     */
+    public List<TransitionModel> getStatusAbfrageEventsBasedOnStateAndAuthorities(final UUID id)
+        throws EntityNotFoundException {
+        List<AuthoritiesEnum> authorities = AuthenticationUtils.getUserAuthorities();
+        List<StatusAbfrageEvents> possibleAbfrageEventsBasedOnAuthorities;
+        Map<AuthoritiesEnum, StatusAbfrageEvents> authoritiesAndMatchingEvent = getAuthoritiesAndEventsMap();
```

Comment:
Da die Variable `authoritiesAndMatchingEvent ` in dieser Methode nur einmal bei Methode `getStatusAbfrageEventsForAuthorities` verwendet wird, kann der Methodenaufruf `getAuthoritiesAndEventsMap` zum instantiieren der Variable in der Methode`getStatusAbfrageEventsForAuthorities` werden.